debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -478,11 +481,68 @@ protected Object extractValueFromMessage(PgProto.DatumMessage datumMessage) {
             }
             case PgOid.TSTZRANGE_OID:
                 return datumMessage.hasDatumBytes() ? new String(datumMessage.getDatumBytes().toByteArray(), Charset.forName("UTF-8")) : null;
+            case PgOid.INT2_ARRAY:
+            case PgOid.INT4_ARRAY:
+            case PgOid.INT8_ARRAY:
+            case PgOid.TEXT_ARRAY:
+            case PgOid.NUMERIC_ARRAY:
+            case PgOid.FLOAT4_ARRAY:
+            case PgOid.FLOAT8_ARRAY:
+            case PgOid.BOOL_ARRAY:
+            case PgOid.DATE_ARRAY:
+            case PgOid.TIME_ARRAY:
+            case PgOid.TIMETZ_ARRAY:
+            case PgOid.TIMESTAMP_ARRAY:
+            case PgOid.TIMESTAMPTZ_ARRAY:
+            case PgOid.BYTEA_ARRAY:
+            case PgOid.VARCHAR_ARRAY:
+            case PgOid.OID_ARRAY:
+            case PgOid.BPCHAR_ARRAY:
+            case PgOid.MONEY_ARRAY:
+            case PgOid.NAME_ARRAY:
+            case PgOid.INTERVAL_ARRAY:
+            case PgOid.CHAR_ARRAY:
+            case PgOid.VARBIT_ARRAY:
+            case PgOid.UUID_ARRAY:
+            case PgOid.XML_ARRAY:
+            case PgOid.POINT_ARRAY:
+            case PgOid.JSONB_ARRAY:
+            case PgOid.JSON_ARRAY:
+            case PgOid.REF_CURSOR_ARRAY:
+                // Currently the logical decoding plugin sends unhandled types as a byte array containing the string
+                // representation (in Postgres) of the array value.
+                // The approach to decode this is sub-optimal but the only way to improve this is to update the plugin.
+                // Reasons for it being sub-optimal include:
+                // 1. It requires a Postgres JDBC connection to deserialize
+                // 2. The byte-array is a serialised string but we make the assumption its UTF-8 encoded (which it will
+                //    be in most cases)
+                // 3. For larger arrays and especially 64-bit integers and the like it is less efficient sending string
+                //    representations over the wire.
+                try {
+                    byte[] data = datumMessage.hasDatumBytes()? datumMessage.getDatumBytes().toByteArray() : null;
```

Comment:
Yes, those are not currently supported by my work. I am not sure if the Postgres JDBC metadata returns the dimensionality of arrays (I would doubt it does) so we would probably need to issue another SQL query to get this info when refreshing schema.